src/manifest/mod.rs

```diff
@@ -187,13 +187,30 @@ pub trait Manifest {
 ///
 /// These types are only really intended to be used as type parameters.
 pub mod provenance {
+    /// A provenance. This trait can be used to write code that is generic
+    /// on provenances but which might choose to skip certain operations for
+    /// the `Adhoc` provenenace.
+    pub trait Provenance {
+        /// Whether this provenance is "trusted", i.e., whether it represents
+        /// a continuous chain of trust up to a signature.
+        const TRUSTED: bool;
```

Comment:
I'm not 100% on board with `TRUSTED` as the designator here. I _trust_ that an adhoc `Provenance` comes from nowhere. Possible followup but I'm OK with this for now.

Reply:
Changed to `AUTHENTICATED` per our discussion.